### DISCUSSION TITLE
get_provider_info validation schema does not validate required properties for connection-types objects

### DISCUSSION THREAD

**Discussion by EugeneMSOff** [apache/airflow#32741] [Q&A] [answered]
As I see, connection-types objects in `get_provider_info` function from custom provider must contains two required properties: "connection-type" and "hook-class-name".

But validation schema from airflow/airflow/provider_info.schema.json does not require it properly.
I think, that problem is here https://github.com/apache/airflow/blob/49921763eb15f68f91da826a86690ba4c4155c35/airflow/provider_info.schema.json#L28-L47

Looks like that require properties must be placed at [this](https://github.com/apache/airflow/blob/49921763eb15f68f91da826a86690ba4c4155c35/airflow/provider_info.schema.json#L42) level (line 42)

Looks like a bug in validation schema, but I'm not sure. 
Make it clease please.

**Comment by Adaverse** [comment, marked as answer]
Nice catch. #32754 should fix this.
